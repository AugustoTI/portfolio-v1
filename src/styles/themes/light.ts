import { Lora, Roboto } from '@next/font/google'

const roboto = Roboto({
  weight: ['400', '500'],
  subsets: ['latin'],
  display: 'swap',
})

const lora = Lora({
  weight: ['400', '500', '600'],
  subsets: ['latin'],
  display: 'swap',
})

export const light = {
  font: {
    sizes: {
      biggest: 'var(--biggest-font-size)',
      h1: 'var(--h1-font-size)',
      h2: 'var(--h2-font-size)',
      h3: 'var(--h3-font-size)',
      normal: 'var(--normal-font-size)',
      small: 'var(--small-font-size)',
      smaller: 'var(--smaller-font-size)',
      tiny: 'var(--tiny-font-size)',
    },
    family: {
      roboto: roboto.style.fontFamily,
      lora: lora.style.fontFamily,
    },
  },
  colors: {
    first: 'hsl(207, 65%, 65%)',
    title: 'hsl(207, 4%, 16%)',
    text: 'hsl(207, 4%, 28%)',
    textLight: 'hsl(207, 4%, 56%)',
    body: 'hsl(207, 4%, 99%)',
    container: 'hsl(207, 4%, 95%)',
  },
  zIndex: {
    normal: 1,
    tooltip: 10,
    fixed: 100,
  },
  height: {
    header: '5.6rem',
  },
  medias: {
    mc: '(max-width:340px)',
    sm: '(min-width:576px)',
    md: '(min-width:767px)',
    lg: '(min-width:1023px)',
    xl: '(min-width:1200px)',
  },
} as const
