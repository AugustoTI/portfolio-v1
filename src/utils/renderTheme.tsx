import { ThemeProvider } from '@emotion/react'
import { render } from '@testing-library/react'
import { ReactNode } from 'react'
import { light } from '~/styles/themes/light'

export const renderTheme = (children: ReactNode) => {
  return render(<ThemeProvider theme={light}>{children}</ThemeProvider>)
}
