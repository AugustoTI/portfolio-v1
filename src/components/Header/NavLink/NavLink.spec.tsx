import { screen } from '@testing-library/react'
import { matchers } from '@emotion/jest'
import userEvent from '@testing-library/user-event'
import { RiHome5Line } from 'react-icons/ri'
import { NavLink } from '.'
import { light } from '~/styles/themes/light'
import { renderTheme } from 'src/utils/renderTheme'

expect.extend(matchers)

describe('<NavLink />', () => {
  it('should render correctly', () => {
    renderTheme(<NavLink href="#">Example</NavLink>)

    const link = screen.getByRole('link', { name: /Example/i })

    expect(link).toBeInTheDocument()
  })

  it('should call the function on click', async () => {
    const mockFn = jest.fn()
    renderTheme(
      <NavLink href="#" onClick={mockFn}>
        Example
      </NavLink>,
    )

    const link = screen.getByRole('link', { name: /Example/i })

    await userEvent.click(link)

    expect(mockFn).toHaveBeenCalled()
    expect(mockFn).toHaveBeenCalledTimes(1)
  })

  it('should change the color when hovering the mouse over it', () => {
    renderTheme(<NavLink href="#">Example</NavLink>)

    const link = screen.getByRole('link', { name: /Example/i })

    expect(link).toHaveStyleRule('color', light.colors.title, {
      target: ':hover',
    })
  })

  it('should change font size on larger screens', () => {
    renderTheme(<NavLink href="#">Example</NavLink>)

    const link = screen.getByRole('link', { name: /Example/i })

    expect(link).toHaveStyleRule('font-size', light.font.sizes.normal, {
      media: light.medias.lg,
    })
  })

  it('icon should disappear on larger screens', () => {
    renderTheme(
      <NavLink href="#">
        <RiHome5Line /> Example
      </NavLink>,
    )

    const link = screen.getByRole('link', { name: /Example/i })

    expect(link).toHaveStyleRule('font-size', '2rem', {
      target: 'svg',
    })

    expect(link).toHaveStyleRule('display', 'none', {
      media: light.medias.lg,
      target: 'svg',
    })
  })

  it('should to match snapshot', () => {
    const { container } = renderTheme(
      <NavLink href="#">
        <RiHome5Line /> Example
      </NavLink>,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
