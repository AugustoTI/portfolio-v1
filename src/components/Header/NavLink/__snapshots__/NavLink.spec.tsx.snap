// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> should to match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  row-gap: 0.4rem;
  font-size: var(--smaller-font-size);
  color: hsl(207, 4%, 56%);
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.emotion-0:hover {
  color: hsl(207, 4%, 16%);
}

.emotion-0 svg {
  font-size: 2rem;
}

@media screen and (min-width:1023px) {
  .emotion-0 {
    font-size: var(--normal-font-size);
  }

  .emotion-0 svg {
    display: none;
  }
}

<a
  class="emotion-0 emotion-1"
  href="#"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M13 19h6V9.978l-7-5.444-7 5.444V19h6v-6h2v6zm8 1a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1V9.49a1 1 0 0 1 .386-.79l8-6.222a1 1 0 0 1 1.228 0l8 6.222a1 1 0 0 1 .386.79V20z"
      />
    </g>
  </svg>
   Example
</a>
`;
